travel-planner/travel-planner-backend/routes/maps.js

```diff
@@ -8,17 +8,43 @@ router.use(bodyParser.json());
 const Parse = require("parse/node");
 
 router.post("/newMap", async (req, res) => {
-	let title = req.body.title;
-	let center = req.body.center;
-	let sessionToken = req.body.sessionToken;
+	try {
+		let title = req.body.title;
+		let center = req.body.center;
+		let sessionToken = req.body.sessionToken;
 
-	// Pass in center and get lat and lng here:
-	// Use this temporarily since getAddress is broken.
-	const lat = 37.354559;
-	const lng = -121.883844;
+		// Pass in center and get lat and lng here:
+		// Use this temporarily since getAddress is broken.
+		const lat = 30;
+		const lng = 0;
 
-	try {
-		mapsModel.createNewMap(title, { lat: lat, lng: lng }, sessionToken);
+		const Maps = Parse.Object.extend("Maps");
+		const Timelines = Parse.Object.extend("Timelines");
+
+		const map = new Maps();
+
+		let query = new Parse.Query("_Session");
+		query.equalTo("sessionToken", sessionToken);
+		query.first().then(function (session) {
+			if (session) {
+				const user = session.get("user"); // sometimes I get an error... ParseError: User is required. Don't know what's causing this though...
```

Comment:
Is it because the user field can be missing (like nullable) in session? How about check if the user exists before go here or give it a fallback value

Reply:
I don't think this is possible since the "user" field is required in the Sessions table in the database. All sessions when created have a user associated with it. I will add a conditional to check if user exists after this line though just in case.

Reply:
Yeah, it shouldn't be missing or nullable. But it could happen due to connection (fail to fetch it)